### ISSUE TITLE
GraphQL API 400 Bad Request

### ISSUE THREAD

**Issue by fuckup1337** [SecurityRiskAdvisors/VECTR#142]
Hi guys,

I am trying to communicate with the API. But for some reason I am getting only 400 errors. 

So far I tried Curl, Burp and psh, but it always fails. 

I am using the format mentioned in the docs:
BurpSuite:
POST /sra-purpletools.rest/graphql
Host: < hostname >
Accept: application/json
Authorization: VEC1 < KeyID >:< Secret >

{
"query": "testcases(db: \"DBNAME\") { nodes {id, name}}
}

Do I need to enable it somewhere else? Or is there something I am missing. 

Thanks!

**Comment by doodleincode** [comment]
I'm not sure if the sample you gave is what you're actually sending, but be sure to escape quotes in your JSON payload:

```
{
  "query": "testcases(db: \"DBNAME\") { nodes {id, name}}"
}
```

There's a known bug in our GraphQL API that is expecting the `variables` key in the JSON payload. This will be fixed in a future release, but for the time being, you can just include an empty `variables` in your payload:

```
{
  "query": "testcases(db: \"DBNAME\") { nodes {id, name}}",
  "variables": {}
}
```

**Comment by carlvonderheid** [comment]
Fixed in VECTR 7.1.9
